packages/react/src/hooks/useTrack.ts

```diff
@@ -9,6 +10,7 @@ export interface UseTrackOptions {
 }
 
 /** @public */
-export function useTrack(trackRef: TrackIdentifier, options: UseTrackOptions = {}) {
-  return useMediaTrackBySourceOrName(trackRef, options);
+export function useTrack(source: Track.Source, participant?: Participant) {
```

Comment:
Non-deterministic if there are multiple tracks with the same source. You pointed out that this is an edge case or only for advanced users, but I'm not a fan of hiding the underlying problem for the sake of not returning an array of tracks. I think this problem applies to useTrack as well.

Reply:
do you haven an alternative suggestion for using a single track?

Reply:
I think 
```tsx
const cameraTracks = useParticipantTracks([Track.Source.Camera], 'participant-id')

if(cameraTracks.length > 0){
  const cameraTrack = cameraTracks[0]
}
```

is the best we can currently offer. I think only if we have something like useTrackById could we return a single track.

Reply:
I understand your point, but I would argue that users expect such an API as they can do the same on all the other client SDKs (i.e. `participant.getTrack(Track.SourceCamera)` and get a single track back). While I hear you saying that we don't have to replicate that behaviour, for me, it's more a question of how to minimise the mental overhead juggling different APIs/SDKs

Reply:
I think most developers will only be exposed to one SDK, so I'm not sure if this is a big problem. On top of that, react components are the first of its kind (component package), so I don't think we have any need to replicate the APIs from the underlying package.
I'm not going to die on this hill, just to summarise things:
The hook is syntactic sugar to make an already simple thing even simpler. If there were no downside, I would be totally fine with it. I personally would not want to sacrifice correctness and predictability for a slight improvement in developer experience.

Reply:
removed `useTrack` for now in order to get a release out quickly. We can decide to re-introduce it later